hepstats/hypotests/calculators/frequentist_calculator.py

```diff
@@ -0,0 +1,430 @@
+
+                >>> calc = FrequentistCalculator(input=loss, minimizer=MinuitMinimizer(), ntoysnull=1000, ntoysalt=1000)
+        """
+
+        super(FrequentistCalculator, self).__init__(input, minimizer)
+
+        self._toysnull = {}
+        self._toysalt = {}
+        self._ntoysnull = ntoysnull
+        self._ntoysalt = ntoysalt
+        self._sampler = sampler
+        self._sample = sample
+        self._toys_loss = {}
+        self._printlevel = 1
+
+    @property
+    def ntoysnull(self):
+        """
+        Returns the number of toys loss for the null hypotesis.
+        """
+        return self._ntoysnull
+
+    @property
+    def ntoysalt(self):
+        """
+        Returns the number of toys loss for the alternative hypotesis.
+        """
+        return self._ntoysalt
+
+    @property
+    def printlevel(self):
+        return self._printlevel
+
+    def sampler(self, floatting_params=None, *args, **kwargs):
+        """
+        Create sampler with models.
+
+        Args:
+            floatting_params (list): floating parameter in the sampler
+
+        Example with `zfit`:
+            >>> sampler = calc.sampler(floatting_params=[zfit.Parameter("mean")])
+        """
+        self.set_dependents_to_bestfit()
+        nevents = []
+        for m, d in zip(self.model, self.data):
+            if m.is_extended:
+                nevents.append("extended")
+            else:
+                nevents.append(get_nevents(d))
+
+        return self._sampler(self.model,  nevents, floatting_params, *args, **kwargs)
+
+    def sample(self, sampler, ntoys, parameter=None, value=None):
+        """
+        Returns the samples generated from the sampler for a given value of a parameter
+
+        Args:
+            sampler (list): generator of samples
+            ntoys (int): number of samples to generate
+            parameter (optionnal): floating parameter in the sampler
+            value (optionnal): value of the parameter
+
+        Example with `zfit`:
+            >>> mean = zfit.Parameter("mean")
+            >>> sampler = calc.sampler(floatting_params=[mean])
+            >>> sample = calc.sample(sampler, 1000, mean, 1.2)
+        """
+        return self._sample(sampler, ntoys, parameter, value)
+
+    def toys_loss(self, parameter):
+        """
+        Construct a loss function constructed with a sampler for a given floating parameter
+
+        Args:
+            parameter: floating parameter in the sampler
+        Returns:
+             Loss function
+
+        Example with `zfit`:
+            >>> loss = calc.toys_loss(zfit.Parameter("mean"))
+        """
+        if parameter.name not in self._toys_loss:
+            sampler = self.sampler(floatting_params=[parameter])
+            self._toys_loss[parameter.name] = self.lossbuilder(self.model, sampler)
+        return self._toys_loss[parameter.name]
+
+    def _generate_and_fit_toys(self, poigen, ntoys, poieval, printfreq=0.2):
```

Comment:
From an aesthetical point of view it would look better to have the poiXXX arguments next to each other. I would specify the number of toys before the poiXXX.

Reply:
Change also the list order in the docstring ;-).

Reply:
So _generate_and_fit_toys(self, ntoys, poigen, poieval, printfreq=0.2):

Reply:
Yep, seems more natural :).